### PR TITLE
[flutter_tools] track null safety usage

### LINKED ISSUE #59686: Add usage tracking for --enable-experiment=non-nullable
This could be baked into the existing command usage functionality, but will require an additional custom dimension.  We should also include this in the hot reload/restart analytics.

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Track usage of commands with enable-experiment=non-nullable, as well as hot reload events where this flag is enabled. Fixes https://github.com/flutter/flutter/issues/59686

### CODE REVIEW

**Comment by jonahwilliams** [comment]
FYI @zanderso

File: packages/flutter_tools/lib/src/runner/flutter_command.dart [hunk 1]
```diff
@@ -841,6 +841,10 @@ abstract class FlutterCommand extends Command<void> {
           ...?await usageValues,
           CustomDimensions.commandHasTerminal:
             globals.stdio.hasTerminal ? 'true' : 'false',
+          if (argParser.options.containsKey(FlutterOptions.kEnableExperiment))
+            CustomDimensions.commandUsingNullSafety: stringsArg(FlutterOptions.kEnableExperiment)
+              .contains('non-nullable')
+              .toString()
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/runner/flutter_command.dart:847]: nit: This is getting hard to read.

**Review by jmagman — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/runner/flutter_command.dart (hunk 1 above)
Comment (zanderso) [reply packages/flutter_tools/lib/src/runner/flutter_command.dart:847]: The additional usage parameters should possibly be `Map<CustomDimentions, dynamic>` with a `toString()` before sending over to `package:usage`.

`FlutterCommand` (or elsewhere) could also have some getters for whether experiments are enabled maybe? Other suggestions?

File: packages/flutter_tools/lib/src/reporting/usage.dart [hunk 2]
```diff
@@ -57,6 +57,8 @@ enum CustomDimensions {
   commandResultEventMaxRss,  // cd44
   commandRunAndroidEmbeddingVersion, // cd45
   commandPackagesAndroidEmbeddingVersion, // cd46
+  commandUsingNullSafety, // cd47
```
[the remark this answers is not in the document]
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/reporting/usage.dart:60]: No these don't ever overlap, using the same value SGTM

File: packages/flutter_tools/lib/src/runner/flutter_command.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/runner/flutter_command.dart]: Added a private List<String> enabledExperiments and simplified the check. Updated to use Object and convert to String in the same place that the enums are converted

**Review by zanderso — APPROVED** [review]

**Comment by zanderso** [comment]
Remember to create the new custom dimension in GA =)

**Pull request merged**